homeassistant/components/modbus/modbus.py

```diff
@@ -40,32 +40,35 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def modbus_setup(
+async def async_modbus_setup(
     hass, config, service_write_register_schema, service_write_coil_schema
 ):
     """Set up Modbus component."""
 
     hass.data[DOMAIN] = hub_collect = {}
     for conf_hub in config[DOMAIN]:
-        hub_collect[conf_hub[CONF_NAME]] = ModbusHub(conf_hub)
+        my_hub = ModbusHub(hass, conf_hub)
+        hub_collect[conf_hub[CONF_NAME]] = my_hub
 
         # modbus needs to be activated before components are loaded
         # to avoid a racing problem
-        hub_collect[conf_hub[CONF_NAME]].setup(hass)
+        await my_hub.async_setup()
 
         # load platforms
         for component, conf_key in PLATFORMS:
             if conf_key in conf_hub:
-                load_platform(hass, component, DOMAIN, conf_hub, config)
+                await async_load_platform(hass, component, DOMAIN, conf_hub, config)
```

Comment:
We need to schedule the loading of the platforms as tasks to not risk a deadlock. Please use `hass.async_create_task`.

Reply:
Makes sense, done.

I had to add a second to the timings in the test_thread_lock, to allow the task to complete.